cli/command/node/cmd.go

```diff
@@ -38,6 +41,16 @@ func Reference(ctx context.Context, client apiclient.APIClient, ref string) (str
 		if err != nil {
 			return "", err
 		}
+		if info.Swarm.NodeID == "" {
+			// If there's no node ID in /info, the node probably
+			// isn't a manager. Call a swarm-specific endpoint to
+			// get a more specific error message.
+			_, err = client.NodeList(ctx, types.NodeListOptions{})
```

Comment:
It was also out of interest, because I saw the multiple implementations, and was hoping we could have a single "here's what to use" function.

If `info.Swarm.ControlAvailable` is true, but the node is locked, won't the `NodeInspectWithRaw ` API call still fail? https://github.com/docker/docker/blob/b1debf1374ca7be7e7b4d4fcd41eceaf8b7cf153/cli/command/node/ps.go#L71-L75